### PR TITLE
cig-pythia: add cig-pythia package to spack

### PR DESCRIPTION
**Pull request by downloadico**

### CODE REVIEW

File: var/spack/repos/builtin/packages/cig-pythia/package.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class CigPythia(AutotoolsPackage):
```
Comment (becker33) [inline var/spack/repos/builtin/packages/cig-pythia/package.py:9]: I think this should inherit from both `AutotoolsPackage` and `PythonExtension`

File: var/spack/repos/builtin/packages/cig-pythia/package.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    depends_on("mpi", when="+mpi")
+    depends_on("python@3.8:")
+    depends_on("py-pip")
+
+    def configure_args(self):
+        spec = self.spec
+        args = []
+        if "+mpi" in spec:
+            args.append(f"CC={spec['mpi'].mpicc}")
+            args.append(f"CXX={spec['mpi'].mpicxx}")
```
Comment (becker33) [inline var/spack/repos/builtin/packages/cig-pythia/package.py]:
```suggestion
        if "+mpi" in spec:
            args.append("--enable-mpi")
            args.append(f"CC={spec['mpi'].mpicc}")
            args.append(f"CXX={spec['mpi'].mpicxx}")
        else:
            args.append("--disable-mpi")
```

**Review by becker33 — CHANGES_REQUESTED** [review]
I'm not an expert in this package, but I took a dive on the build system and it looks like a python extension. If so, please look at https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#choosing-a-package-name and consider whether the package name should be prefixed with `py-`.

**Comment by downloadico** [comment]
Hi Gregory,

I'm pretty sure that you're right.  I thought that C++ was a larger 
portion and more important part of the code.  I've "stepped in" to build 
another code that depends on this.  The other code is known to need 
consistency with compilers and their flags, libraries, etc. so I thought 
that that sounded like a job for Spack!

The actual name of this code is "Pythia"; I'd chosen cig-pythia to avoid 
conflicts with the existing pythia6 and pythia8 on the assumption that 
there may someday be a "pythia" package for some future releases of the 
other Pythia.  Do you think it would be better to name this package 
py-pythia dropping the "cig" in the name entirely or keep it?  "cig" is 
short for "Computational Infrastructure for Geodynamics" which created 
this package.

Thanks,
Jim

On Wed, 20 Mar 2024, Greg Becker wrote:

> @becker33 requested changes on this pull request.
>
> I'm not an expert in this package, but I took a dive on the build system and it looks like a python extension. If so, please look at https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#choosing-a-package-name and consider whether the package name should be prefixed with `py-`.
>
>> @@ -0,0 +1,41 @@
> +# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
> +# Spack Project Developers. See the top-level COPYRIGHT file for details.
> +#
> +# SPDX-License-Identifier: (Apache-2.0 OR MIT)
> +
> +from spack.package import *
> +
> +
> +class CigPythia(AutotoolsPackage):
>
> I think this should inherit from both `AutotoolsPackage` and `PythonExtension`
>
>> +        if "+mpi" in spec:
> +            args.append(f"CC={spec['mpi'].mpicc}")
> +            args.append(f"CXX={spec['mpi'].mpicxx}")
>
> ```suggestion
>        if "+mpi" in spec:
>            args.append("--enable-mpi")
>            args.append(f"CC={spec['mpi'].mpicc}")
>            args.append(f"CXX={spec['mpi'].mpicxx}")
>        else:
>            args.append("--disable-mpi")
> ```
>
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/spack/spack/pull/43247#pullrequestreview-1949394568
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>

James E. Prewett                    ***@***.*** ***@***.***
Systems Team Leader           LoGS: http://www.hpc.unm.edu/~download/LoGS/
Designated Security Officer         OpenPGP key: pub 1024D/31816D93
HPC Systems Engineer III   UNM HPC  505.277.8210

**Pull request closed without merging**